main/src/main/scala/sbt/internal/server/LanguageServerReporter.scala

```diff
@@ -51,7 +51,7 @@ class LanguageServerReporter(
     val pos = problem.position
     pos.sourceFile.toOption foreach { sourceFile: File =>
       problemsByFile.get(sourceFile) match {
-        case Some(xs: List[Problem]) => problemsByFile(sourceFile) = problem :: xs
+        case Some(xs: List[Problem]) => problemsByFile(sourceFile) = xs :+ problem
```

Comment:
As Guillaume noted on Gitter, could you replace this a better data structure for appending?

Reply:
Back from holydays, sure I'll do it.

Reply:
Ok I rebased and pushed a new commit using `ListBuffer`